src/session/content/chat_history.rs

```diff
@@ -287,15 +289,22 @@ impl ChatHistory {
                 chat.history().to_owned().upcast()
             };
 
-            let selection = gtk::NoSelection::new(Some(&chat_history));
+            let chat_history = chat.history();
+            spawn(clone!(@weak chat_history => async move {
+                while chat_history.n_items() < MIN_N_ITEMS {
+                    let limit = MIN_N_ITEMS - chat_history.n_items();
+                    if chat_history.load_older_messages(limit as i32).await.is_err() {
+                        break;
+                    }
```

Comment:
Is this the right way?
I think we should at least log the error.
Then, I think, we should distinguish between our error variants. In case of a `AlreadyLoading` we should at least check again after some time if still more messages have to be loaded.

Reply:
> I think we should at least log the error in case of an TDLib error.

Yeah, that's true.

> In case of a AlreadyLoading we should at least check again after some time if still more messages have to be loaded.

Mh, the problem though is that when we call this function here, we expect that the history is not loading, otherwise something is off. I think we should also warn for this case.

Reply:
Maybe I'm wrong, but AFAIS, it is still possible that the user scrolls up and triggers a message load while we are still in the while loop. Scrolling may happen across the `.await` point inside the while loop.

And the greater `MIN_N_ITEMS` is, the more likely this becomes.

Maybe this is not a real problem because a prerequisite for scrolling is that the history is filled with enough messages. But we could end up in a situation where the contsraint `messages` >= `MIN_N_ITEMS` is violated.

Reply:
So I guess just logging the error would be enough for now.

Reply:
> Oh never mind, I've just realized that the message loading of the scroll action would then error out.
> So I guess just logging the error would be enough for now.

Yeah, I'm pretty sure that the logging is fine here.